data/abi/abi_encode_test.go

```diff
@@ -323,50 +282,70 @@ func TestEncodeValid(t *testing.T) {
 	   0x00, 0x00      (first dynamic bool array length 0)
 	   0x00, 0x00      (second dynamic bool array length 0)
 	*/
+	tupleType, err = TypeFromString("(bool[],bool[])")
+	require.NoError(t, err, "type from string for tuple type should not return error")
 	t.Run("empty dynamic array tuple encoding", func(t *testing.T) {
-		emptyDynamicArray, err := MakeDynamicArray([]Value{}, MakeBoolType())
-		require.NoError(t, err, "make empty dynamic array should not return error")
-		tupleVal, err := MakeTuple([]Value{emptyDynamicArray, emptyDynamicArray})
-		require.NoError(t, err, "make empty dynamic array tuple should not return error")
 		expected := []byte{
 			0x00, 0x04, 0x00, 0x06,
 			0x00, 0x00, 0x00, 0x00,
 		}
-		actual, err := tupleVal.Encode()
+		actual, err := tupleType.Encode([]interface{}{
+			[]bool{}, []bool{},
+		})
 		require.NoError(t, err, "encode empty dynamic array tuple should not return error")
 		require.Equal(t, expected, actual, "encode empty dynamic array tuple does not match with expected")
 	})
 
 	// encoding test for empty tuple
 	// input: (), expected encoding: ""
+	tupleType, err = TypeFromString("()")
+	require.NoError(t, err, "type from string for tuple type should not return error")
 	t.Run("empty tuple encoding", func(t *testing.T) {
-		emptyTuple, err := MakeTuple([]Value{})
-		require.NoError(t, err, "make empty tuple should not return error")
 		expected := make([]byte, 0)
-		actual, err := emptyTuple.Encode()
+		actual, err := tupleType.Encode([]interface{}{})
 		require.NoError(t, err, "encode empty tuple should not return error")
 		require.Equal(t, expected, actual, "empty tuple encode should not return error")
 	})
 }
 
+func castBigIntToInterface(val *big.Int, bitSize uint16) (interface{}, error) {
```

Comment:
I'm not seeing every place where this is used, but it seems dangerous.  IT's inferring the type based one the size of its arguments?  So I can't get a val.Uint64() from a small number.   If that's a good idea in some contexts, I'd at least prefer a name like "castBigIntToNarrowest()" or similar.

I suppose it's not so dangerous since it's only in testing code.

Reply:
I discarded the function now, don't think it would be a nice idea to use it...